packages/domain-picker/src/domain-picker/style.scss

```diff
@@ -177,6 +176,13 @@
 	flex-grow: 1;
 	letter-spacing: 0.4px;
 
+	.domain-picker__suggestion-item-name-inner {
+		// vertically align elements within, needed for items showing information icon.
+		display: flex;
+		align-items: center;
```

Comment:
There's still a bit of mis-alignment since the button is 27px high and icon 24px, and icon isn't center-aligned within the button.

<img width="990" alt="image" src="https://user-images.githubusercontent.com/87168/92895310-0f3edf00-f424-11ea-83a9-bdb936c705a0.png">
<img width="411" alt="image" src="https://user-images.githubusercontent.com/87168/92895374-1cf46480-f424-11ea-94db-406c04c9b4d2.png">

[It's not documented](https://github.com/WordPress/gutenberg/tree/7d1bcdf4ac1b3970bd43bf5e33a05fe2b7c3cd81/packages/components/src/button#readme) (not sure why), but I wonder if this would help: `<Button icon={ info } />`